# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/UseCounters.conf
# Commit: 5a65d2960e0d
# Full Hash: 5a65d2960e0d088d5f9b7ec573e6970014b58687
# Author: Matthew Gaudet <mgaudet@mozilla.com>
# Date: 2025-05-16 09:19:19
# Regressor Bug: 1964451
# File Overlap Count: 1
# Description:
#   Bug 1964451 - Add use counters for OOM handling r=sfink,mccr8
#   
#   So the goal here is that we get OOM data where processes survive long enough
#   send telemetry. To get the case where processes -crash- we'll also have to
#   get crash data. See Bug 1964092 for that half.
# ==============================================================================

diff -r 2f8d1dbbacf6 -r 5a65d2960e0d dom/base/UseCounters.conf
--- a/dom/base/UseCounters.conf	Thu May 15 19:59:25 2025 +0000
+++ b/dom/base/UseCounters.conf	Thu May 15 20:04:27 2025 +0000
@@ -85,6 +85,10 @@
 custom JS_dateparse called Date.parse
 custom JS_dateparse_impl_def call to Date.parse used implementation defined behaviour
 custom JS_regexp_symbol_protocol_on_primitive passed a primitive with a custom implementation of the regexp protocol symbols to one of the String methods that accept a regexp
+custom JS_large_oom_reported reported a large oom
+custom JS_small_oom_reported reported a small oom
+custom JS_large_oom_recovered recovered a large oom
+custom JS_small_oom_recovered recovered a small oom
 
 // Console API
 method console.assert